### PR TITLE
Recurring snapshot/backup job supports imagePullSecrets

### PR DESCRIPTION
**Pull request by jenting**
https://github.com/longhorn/longhorn/issues/2432

Signed-off-by: JenTing Hsiao <jenting.hsiao@suse.com>

### CODE REVIEW

File: controller/recurring_job_controller.go [hunk 1]
```diff
@@ -377,6 +381,11 @@ func (control *RecurringJobController) newCronJob(recurringJob *longhorn.Recurri
 							Tolerations:        util.GetDistinctTolerations(tolerations),
 							NodeSelector:       nodeSelector,
 							PriorityClassName:  priorityClass.Value,
+							ImagePullSecrets: []corev1.LocalObjectReference{
+								{
+									Name: registrySecretSetting.Value,
```
Comment (innobead) [inline controller/recurring_job_controller.go:386]: If the setting is empty, what happens? should we ignore the empty case here?

Comment (jenting) [reply controller/recurring_job_controller.go]: Yes, we should. Thanks for the reminder.

**Review by shuo-wu — COMMENTED** [review]
Should we check all old recurring job CRs in the upgrade path? have a correction function in this controller? Besides, there are recurring job CR creation `translateStorageClassRecurringJobs` in `v110tov120/upgrade.go`.

**Comment by jenting** [comment]
> Should we check all old recurring job CRs in the upgrade path? have a correction function in this controller? Besides, there are recurring job CR creation `translateStorageClassRecurringJobs` in `v110tov120/upgrade.go`.

Probably no, because it means that the old Longhorn cluster recurring job did not work because Longhorn doesn't support `imagePullSecret`.
So I think the user did not use the private registry + private registry secret + recurring job in the end.

But it's nice to have to auto recreate the old recurring job.

**Comment by shuo-wu** [comment]
Does it not work in v1.1.x as well? If No, we can skip considering the upgrade path. Otherwise, it's better to add it.

**Comment by jenting** [comment]
The original issue reported is v1.1.0, so I believe it's not working.

**Review by shuo-wu — APPROVED** [review]

**Pull request merged**